chains/substrate/events.go

```diff
@@ -28,57 +30,250 @@ var Subscriptions = []struct {
-	evt, ok := evtI.(events.EventGenericTransfer)
-	if !ok {
-		return msg.Message{}, fmt.Errorf("failed to cast EventGenericTransfer type")
+	metadata, err := getFieldValueAsByteSlice("Vec<u8>", eventFields)
+	if err != nil {
+		return msg.Message{}, err
 	}
 
-	log.Info("Got generic transfer event!", "destination", evt.Destination, "resourceId", evt.ResourceId)
+	log.Info("Got generic transfer event!", "destination", chainID, "resourceId", fmt.Sprintf("%x", resourceID))
 
 	return msg.NewGenericTransfer(
 		0, // Unset
-		msg.ChainId(evt.Destination),
-		msg.Nonce(evt.DepositNonce),
-		msg.ResourceId(evt.ResourceId),
-		evt.Metadata,
+		msg.ChainId(chainID),
+		msg.Nonce(depositNonce),
+		resourceID,
+		metadata,
 	), nil
 }
+
+func to32Bytes(array []types.U8) ([32]byte, error) {
+	var res [32]byte
+
+	if len(array) != 32 {
+		return res, errors.New("array length mismatch")
+	}
+
+	for i, item := range array {
+		res[i] = byte(item)
+	}
+
+	return res, nil
+}
+
+func getFieldValueAsType[T any](fieldName string, eventFields map[string]any) (T, error) {
+	var t T
+
+	for name, value := range eventFields {
+		if name == fieldName {
+			if v, ok := value.(T); ok {
+				return v, nil
+			}
+
+			return t, fmt.Errorf("field type mismatch, expected %T, got %T", t, value)
+		}
+	}
+
+	return t, fmt.Errorf("field with name '%s' not found", fieldName)
+}
+
+func getFieldValueAsSliceOfType[T any](fieldName string, eventFields map[string]any) ([]T, error) {
+	for name, value := range eventFields {
+		if name == fieldName {
+			value, ok := value.([]any)
+
+			if !ok {
+				return nil, errors.New("field value not an array")
+			}
+
+			res, err := convertSliceToType[T](value)
+
+			if err != nil {
+				return nil, err
+			}
+
+			return res, nil
+		}
+	}
+
+	return nil, fmt.Errorf("field with name '%s' not found", fieldName)
+}
+
+func getFieldValueAsByteSlice(fieldName string, eventFields map[string]any) ([]byte, error) {
+	for name, value := range eventFields {
+		if name == fieldName {
+			value, ok := value.([]any)
+
+			if !ok {
+				return nil, errors.New("field value not an array")
+			}
+
+			slice, err := convertSliceToType[types.U8](value)
+
+			if err != nil {
+				return nil, err
+			}
+
+			res, err := convertToByteSlice(slice)
+
+			if err != nil {
+				return nil, err
+			}
+
+			return res, nil
+		}
+	}
+
+	return nil, fmt.Errorf("field with name '%s' not found", fieldName)
+}
+
+func convertSliceToType[T any](array []any) ([]T, error) {
+	res := make([]T, 0)
+
+	for _, item := range array {
+		if v, ok := item.(T); ok {
+			res = append(res, v)
+			continue
+		}
+
+		var t T
+
+		return nil, fmt.Errorf("couldn't cast '%T' to '%T'", item, t)
+	}
+
+	return res, nil
+}
+
+func convertToByteSlice(array []types.U8) ([]byte, error) {
+	res := make([]byte, 0)
+
+	for _, item := range array {
+		res = append(res, byte(item))
+	}
+
+	return res, nil
+}
+
+func getAmount(eventFields map[string]any) (types.U256, error) {
```

Comment:
We can prob name it as:
```suggestion
func getU256(eventFields map[string]any) (types.U256, error) {
```
if every u256 is always a [u64:4], in any case this is just a hack so no issues here 👍